model/registry/test.yaml

```diff
@@ -0,0 +1,104 @@
+groups:
+  - id: registry.test.suite
+    prefix: test
+    type: attribute_group
+    brief: >
+     This group describes attributes specific to
+     [software tests](https://en.wikipedia.org/wiki/Software_testing).
+    attributes:
+      - id: suite.name
+        type: string
+        stability: experimental
+        brief: >
+          The human readable name of a [test suite](https://en.wikipedia.org/wiki/Test_suite).
+        examples: ['TestSuite1']
+      - id: suite.run.id
+        type: string
+        stability: experimental
+        brief: >
+          The unique identifier of a test suite run.
+        examples: ['123408']
+      - id: suite.run.status
+        type:
+          allow_custom_values: true
+          members:
+            - id: success
+              value: success
+              brief: success
+              stability: experimental
+            - id: failure
+              value: failure
+              brief: failure
+              stability: experimental
+            - id: skipped
+              value: skipped
+              brief: skipped
+              stability: experimental
+            - id: aborted
+              value: aborted
+              brief: aborted
+              stability: experimental
+            - id: timed_out
+              value: timed_out
+              brief: timed_out
+              stability: experimental
+        stability: experimental
+        brief: >
+          The status of the test suite run.
+        examples: ['success', 'failure', 'skipped', 'aborted', 'timed_out']
+      - id: suite.case.name
+        type: string
+        stability: experimental
+        brief: >
+          The human readable name of the [test case](https://en.wikipedia.org/wiki/Test_case).
+        examples: ['TestCase1']
+      - id: suite.case.run.id
+        type: string
+        stability: experimental
+        brief: >
+          The unique identifier of a test case run.
+        examples: ['123445']
+      - id: suite.case.run.status
```

Comment:
they are repeating `suite.run.status`. What about having it defined only once?

Reply:
A test case run can have a different status than a test suite run & there can be many test case runs within a suite of tests. Because of that, I opted for providing a status for the run of the case in addition to the run of the test suite.

Reply:
I do understand this but in current implementation they are the same. And we allow to define custom values for al enums. If you will provide different existing enum values then it will be ok

Reply:
I just realized that I was looking at the wrong attribute. There are three status attributes that I have defined right now:

* suite.run.status ( the status of the entire run for the test suite ) 
* case.run.status ( the status of the run for the test case within the suite ) 
* case.execution.status ( the pass / fail result of the test case that was run - different than the )

The use of `run` is what denotes the differences for status. `Run` is referring to the actual run of the tests within a pipeline, and not necessarily the test execution itself. That's the differentiation.